src/librustc/hir/mod.rs

```diff
@@ -934,7 +934,8 @@ pub struct BodyId {
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Body {
     pub arguments: HirVec<Arg>,
-    pub value: Expr
+    pub value: Expr,
+    pub is_generator: bool,
```

Comment:
Why do we have both `is_generator: bool` and `is_generator: IsGenerator`? Maybe we should have this be `BodyKind` and also have consts in the mix (we need to check how that looks).